src/Features/CSharp/Portable/QuickInfo/CSharpSyntacticQuickInfoProvider.cs

```diff
@@ -28,20 +30,38 @@ public CSharpSyntacticQuickInfoProvider()
         protected override Task<QuickInfoItem?> BuildQuickInfoAsync(
             QuickInfoContext context,
             SyntaxToken token)
-            => Task.FromResult(BuildQuickInfo(token));
+            => Task.FromResult(BuildQuickInfo(token, context.CancellationToken));
 
         protected override Task<QuickInfoItem?> BuildQuickInfoAsync(
             CommonQuickInfoContext context,
             SyntaxToken token)
-            => Task.FromResult(BuildQuickInfo(token));
+            => Task.FromResult(BuildQuickInfo(token, context.CancellationToken));
 
-        private static QuickInfoItem? BuildQuickInfo(SyntaxToken token)
+        private static QuickInfoItem? BuildQuickInfo(SyntaxToken token, CancellationToken cancellationToken)
         {
-            if (token.Kind() != SyntaxKind.CloseBraceToken)
+            switch (token.Kind())
             {
-                return null;
+                case SyntaxKind.CloseBraceToken:
+                    return BuildQuickInfoCloseBrace(token);
+                case SyntaxKind.HashToken:
+                case SyntaxKind.EndRegionKeyword:
+                case SyntaxKind.EndIfKeyword:
+                case SyntaxKind.ElseKeyword:
+                case SyntaxKind.ElifKeyword:
+                case SyntaxKind.EndOfDirectiveToken:
+                    return BuildQuickInfoDirectives(token, cancellationToken);
+                case SyntaxKind.EndOfFileToken:
```

Comment:
when is this case hit?

Reply:
e.g. in some of the RegionEndShowsStartRegionMessageAtDifferentPositions cases:
```CS
#region Start
#endregion$$
```
But I would like to remove that case because I don't thinks it is worth the complications.